File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        var left = 0;
        var right = nums.length - 1;
        var mid = left + right / 2;

        while (left <= right) {
            if (nums[mid] == target) {
                return mid;
            }
            
            if ((nums[left] < nums[mid] && target >= nums[left] && target < nums[mid]) 
                || (nums[left] > nums[mid] && (target > nums[right] || target < nums[mid]))) {
                right = mid;
                mid = (right + left) / 2;
            } else {
                left = mid + 1;
                mid = (right + left) / 2;
            }
        }

        return -1;
    }
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        
        def dfs(node, acc):
            if node.left:
                res = dfs(node.left, acc)
                if res:
                    return res

            acc["count"] -= 1

            if acc["count"] == 0:
                return node.val

            if node.right:
                res = dfs(node.right, acc)
                if res:
                    return res

        return dfs(root, { "count": k })